frontend/test/metabase/scenarios/admin/settings/public-sharing.cy.spec.js

```diff
@@ -0,0 +1,236 @@
+        "have.attr",
+        "href",
+        `/dashboard/${dashboardId}-${expectedDashboardSlug}`,
+      );
+    });
+
+    cy.button("Revoke link").click();
+    modal().within(() => {
+      cy.findByText("Disable this link?").should("be.visible");
+      cy.button("Yes").click();
+    });
+    cy.findByText("No dashboards have been publicly shared yet.").should(
+      "be.visible",
+    );
+  });
+
+  it("should see public questions", () => {
+    const expectedQuestionName = "Public question";
+    const expectedQuestionSlug = "public-question";
+    cy.createQuestion({
+      name: expectedQuestionName,
+      query: {
+        "source-table": ORDERS_ID,
+      },
+    })
+      .then(({ body }) => {
+        const questionId = body.id;
+        cy.wrap(questionId).as("questionId");
+        cy.request("POST", `/api/card/${questionId}/public_link`, {});
+      })
+      .then(response => {
+        cy.wrap(response.body.uuid).as("questionUuid");
+      });
+
+    cy.visit("/admin/settings/public-sharing");
+
+    cy.findByText("Shared Questions").should("be.visible");
+    cy.findByText(expectedQuestionName).should("be.visible");
+    cy.get("@questionUuid").then(questionUuid => {
+      cy.findByText(new RegExp(questionUuid)).should("be.visible");
+    });
+
+    cy.get("@questionId").then(questionId => {
+      cy.findByText(expectedQuestionName).should(
+        "have.attr",
+        "href",
+        `/question/${questionId}-${expectedQuestionSlug}`,
+      );
+    });
+
+    cy.button("Revoke link").click();
+    modal().within(() => {
+      cy.findByText("Disable this link?").should("be.visible");
+      cy.button("Yes").click();
+    });
+    cy.findByText("No questions have been publicly shared yet.").should(
+      "be.visible",
+    );
+  });
+
+  it("should see public actions", () => {
+    enableActionsForDB();
+    const expectedActionName = "Public action";
+
+    cy.createQuestion(
+      {
+        name: "Question",
+        query: {
+          "source-table": ORDERS_ID,
+        },
+      },
+      {
+        dataset: true,
+      },
+    ).then(({ body }) => {
+      const modelId = body.id;
+      cy.wrap(modelId).as("modelId");
+    });
+
+    cy.get("@modelId").then(modelId => {
+      createAction({
+        name: expectedActionName,
+        model_id: modelId,
+      }).then(({ body }) => {
+        const actionId = body.id;
+        cy.wrap(actionId).as("actionId");
+      });
+    });
+
+    cy.get("@actionId")
+      .then(actionId => {
+        cy.request("POST", `/api/action/${actionId}/public_link`, {});
+      })
+      .then(({ body }) => {
+        cy.wrap(body.uuid).as("actionUuid");
+      });
+
+    cy.visit("/admin/settings/public-sharing");
+
+    cy.findByText("Shared Actions").should("be.visible");
+    cy.findByText(expectedActionName).should("be.visible");
+    cy.get("@actionUuid").then(actionUuid => {
+      cy.findByText(new RegExp(actionUuid)).should("be.visible");
```

Comment:
Why do we need to convert it to a regexp?

Reply:
the public URL is partially matched. I think I might as well construct the whole URL, so I can omit using Regex.